backend/src/schema/resolvers/rewards.js

```diff
@@ -1,47 +1,47 @@
-export default {
-  Mutation: {
-    reward: async (_object, params, context, _resolveInfo) => {
-      const { fromBadgeId, toUserId } = params
-      const session = context.driver.session()
-
-      let transactionRes = await session.run(
-        `MATCH (badge:Badge {id: $badgeId}), (rewardedUser:User {id: $rewardedUserId})
-          MERGE (badge)-[:REWARDED]->(rewardedUser)
-          RETURN rewardedUser {.id}`,
-        {
-          badgeId: fromBadgeId,
-          rewardedUserId: toUserId,
-        },
-      )
+import { neode } from '../../bootstrap/neo4j'
+import { UserInputError } from 'apollo-server'
 
-      const [rewardedUser] = transactionRes.records.map(record => {
-        return record.get('rewardedUser')
-      })
+const instance = neode()
 
-      session.close()
+const getUserAndBadge = async ({ badgeKey, userId }) => {
+  let user = await instance.first('User', 'id', userId)
+  const badge = await instance.first('Badge', 'id', badgeKey)
+  if (!user) throw new UserInputError("Couldn't find a user with that id")
+  if (!badge) throw new UserInputError("Couldn't find a badge with that id")
+  return { user, badge }
+}
 
-      return rewardedUser.id
+export default {
+  Mutation: {
+    reward: async (_object, params, context, _resolveInfo) => {
+      const { user, badge } = await getUserAndBadge(params)
+      await user.relateTo(badge, 'rewarded')
+      return user.toJson()
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-11T14:28:08Z" title="Thursday, July 11th 2019, 4:28:08 pm +02:00">Jul 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,47 +1,47 @@
-export default {
-  Mutation: {
-    reward: async (_object, params, context, _resolveInfo) => {
-      const { fromBadgeId, toUserId } = params
-      const session = context.driver.session()
-
-      let transactionRes = await session.run(
-        `MATCH (badge:Badge {id: $badgeId}), (rewardedUser:User {id: $rewardedUserId})
-          MERGE (badge)-[:REWARDED]->(rewardedUser)
-          RETURN rewardedUser {.id}`,
-        {
-          badgeId: fromBadgeId,
-          rewardedUserId: toUserId,
-        },
-      )
+import { neode } from '../../bootstrap/neo4j'
+import { UserInputError } from 'apollo-server'
 
-      const [rewardedUser] = transactionRes.records.map(record => {
-        return record.get('rewardedUser')
-      })
+const instance = neode()
 
-      session.close()
+const getUserAndBadge = async ({ badgeKey, userId }) => {
+  let user = await instance.first('User', 'id', userId)
+  const badge = await instance.first('Badge', 'key', badgeKey)
+  if (!user) throw new UserInputError("Couldn't find a user with that id")
+  if (!badge) throw new UserInputError("Couldn't find a badge with that key")
+  return { user, badge }
+}
 
-      return rewardedUser.id
+export default {
+  Mutation: {
+    reward: async (_object, params, context, _resolveInfo) => {
+      const { user, badge } = await getUserAndBadge(params)
+      await user.relateTo(badge, 'rewarded')
+      return user.toJson()
```


this is nice succinct code, easy to understand :smile_cat: